### DISCUSSION TITLE
Support mirroring of github releases to an alternate location

### DISCUSSION THREAD

**Discussion by IgorTodorovskiIBM** [zopencommunity/meta#547] [General]
Let's discuss options for supporting installs from an alternate location. This is useful when a z/OS system is airgapped and does not have access to Github.

Considerations:
* We should be able to mirror/sync the github pax.Z releases from Github to another location
  * We would need to add support for local mirrors, artifactory, plus other options.
* We should be able to generate a zopen_releases.json that points to the new mirrored download locations
* We should provide a way for zopen install/init to pull down the custom zopen_releases.json file

It could be configured similar to the apt-mirror tool https://www.howtoforge.com/local_debian_ubuntu_mirror or dnf: https://trivialfis.github.io/linux/2017/02/11/Create-a-local-mirror-with-dnf.html

**Comment by v1gnesh** [comment]
The most boring (and frankly most acceptable for some/many sites) option would be to mimic existing repository mgmt tools they use - Sonatype Nexus Repository or JFrog Artifactory. Because it'll translate directly to steps they can reuse in their own site.
Both of these may have some OSS edition, but they will require investment of time etc. to setup.
However, I think, sadly, it's the easiest way to get adoption even from the late adopters.

One other alternative is https://packagecloud.io/ (now owned by Buildkite :wink:).
